pkg/logs/launchers/file/launcher.go

```diff
@@ -158,21 +178,26 @@ func (s *Launcher) cleanup() {
 	s.oldInfoMap = make(map[string]*oldTailerInfo)
 }
 
-// scan checks all the files we're expected to tail, compares them to the currently tailed files,
+// resolveActiveTailers checks all the files we're expected to tail, compares them to the currently tailed files,
 // and triggers the required updates.
 // For instance, when a file is logrotated, its tailer will keep tailing the rotated file.
 // The Scanner needs to stop that previous tailer, and start a new one for the new file.
-func (s *Launcher) scan() {
-	files := s.fileProvider.FilesToTail(s.validatePodContainerID, s.activeSources, s.registry)
+func (s *Launcher) resolveActiveTailers(files []*tailer.File) {
+	tailersFilesCopy := make([]*tailer.File, len(files))
+	copy(tailersFilesCopy, files)
+
+	// Union so scan doesn't remove files added by addSource but not yet found by FilesToTail
```

Comment:
Can you expand on this comment? Can we describe why scan might remove files added by `addSource`? 
Followup question: Is the opposite possible? can scan add back files that were removed by `removeSource`?

Reply:
Nice thanks for confirming 👍